scan-command/src/main/java/io/ballerina/scan/utils/ScanUtils.java

```diff
@@ -550,4 +555,91 @@ private static Optional<String> loadRemoteJAR(Path targetDir, String fileName, U
             return Optional.empty();
         }
     }
+
+    /**
+     * Prints the {@link List<Rule>} of static analysis rules to the console.
+     *
+     * @param rules        the list of static analysis rules
+     * @param outputStream the print stream
+     */
+    public static void printRulesToConsole(List<Rule> rules, PrintStream outputStream) {
+        int maxRuleIDLength = RULE_ID_COLUMN.length();
+        int maxSeverityLength = RULE_SEVERITY_COLUMN.length();
+        int maxDescriptionLength = RULE_DESCRIPTION_COLUMN.length();
+
+        for (Rule rule : rules) {
+            maxRuleIDLength = Math.max(maxRuleIDLength, rule.id().length());
+            maxSeverityLength = Math.max(maxSeverityLength, rule.kind().toString().length());
+            maxDescriptionLength = Math.max(maxDescriptionLength, rule.description().length());
+        }
+
+        String format = "\t%-" + maxRuleIDLength + "s | %-" + maxSeverityLength + "s | %-" + maxDescriptionLength
+                + "s%n";
+
+        outputStream.printf(format, RULE_ID_COLUMN, RULE_SEVERITY_COLUMN, RULE_DESCRIPTION_COLUMN);
+        outputStream.printf("\t" + "-".repeat(maxRuleIDLength + 1) + "--" +
+                "-".repeat(maxSeverityLength + 1) + "--" + "-".repeat(maxDescriptionLength + 1) + "%n");
+
+        sortRules(rules);
+        for (Rule rule : rules) {
+            String formattedLine = String.format(format, rule.id(), rule.kind().toString(), rule.description());
+            outputStream.println(formattedLine.stripTrailing());
+        }
+    }
+
+    /**
+     * <p> Sorts the rules based on a specified order. </p>
+     * <p> The rules are sorted based on the following order. </p>
+     * <ul>
+     *  <li> -1 if the first element should come before the second element </li>
+     *  <li>  1 if the second element should come before the first element </li>
+     * </ul>
+     * <p> The sorting priority is provided as follows. </p>
+     * <ol>
+     *  <li> `ballerina` rules first </li>
+     *  <li> `ballerina/moduleName` rules next </li>
+     *  <li> `ballerinax/moduleName` rules next </li>
+     *  <li> `wso2/moduleName` rules next </li>
+     *  <li> All remaining rules sorted alphabetically </li>
+     * </ol>
+     *
+     * @param rules The list of rules to be sorted.
+     */
+    private static void sortRules(List<Rule> rules) {
+        Comparator<Rule> ruleComparator = (r1, r2) -> {
+            String id1 = r1.id().split(":")[0];
```

Comment:
I feel like this can be done more cleanly. What you need is a list of prefixes sorted in the priority you need. First check if the prefix is in the list and if so compare by index. Else compare by string values. Otherwise if we want to make changes to the sort order, like adding new stuff, this would be very confusing. For example (change as needed),

```
public int compare(String s1, String s2) {
        // Define the priorities for the prefixes
        List<String> priorities = List.of("ballerina", "ballerina/", "ballerinax/", "wso2/");

        // Determine the priority of each string
        int priority1 = getPriority(s1, priorities);
        int priority2 = getPriority(s2, priorities);

        // Compare based on priority
        if (priority1 != priority2) {
            return Integer.compare(priority1, priority2);
        }

        // If both strings have the same priority, sort alphabetically
        return s1.compareTo(s2);
    }

    private int getPriority(String s, List<String> priorities) {
        return priorities.stream()
            .filter(s::startsWith)
            .findFirst()
            .map(priorities::indexOf)
            .orElse(priorities.size());
    }
```